### PR TITLE
Enhance the ACL related models

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch 

The PR cleans up how related models are referenced and adds two utilities to ACL.

### CODE REVIEW

File: common/models/role-mapping.js [hunk 1]
```diff
@@ -17,6 +17,15 @@ module.exports = function(RoleMapping) {
   RoleMapping.APP = RoleMapping.APPLICATION = 'APP';
   RoleMapping.ROLE = 'ROLE';
 
+  RoleMapping.getRelatedModels = function() {
+    if (!this.userModel) {
```
Comment (ritch) [inline common/models/role-mapping.js]: This is weird... Why can't we just do this in `setup()`? I don't like that we aren't returning anything and just modifying the `RoleMapping` model.

File: common/models/role-mapping.js [hunk 2]
```diff
@@ -24,11 +33,10 @@ module.exports = function(RoleMapping) {
    * @param {Application} application
    */
   RoleMapping.prototype.application = function(callback) {
-    var registry = this.constructor.registry;
+    this.constructor.getRelatedModels();
```
Comment (ritch) [inline common/models/role-mapping.js]: This is really weird...

File: common/models/role.js [hunk 3]
```diff
@@ -31,19 +28,19 @@ module.exports = function(Role) {
     return new Date();
   };
 
+  Role.getRelatedModels = function() {
```
Comment (ritch) [inline common/models/role.js]: same concern as `RoleMapping` version of this... I'm guessing there is something preventing from doing this is a more standard way... Can you elaborate?

File: common/models/scope.js [hunk 4]
```diff
@@ -13,6 +13,13 @@ var loopback = require('../../lib/loopback');
  */
 
 module.exports = function(Scope) {
+  Scope.getRelatedModels = function() {
```
Comment (ritch) [inline common/models/scope.js]: Same concern as elsewhere...

**Comment by ritch** [comment]
I'd like to see the comments addressed here somehow. The rest of the code looks good.

File: common/models/role-mapping.js (hunk 1 above)
Comment (raymondfeng) [inline common/models/role-mapping.js]: `setup` is still too early to resolve the related models as they are guaranteed to be available in the registry when `setup` is called.

I'll rename the method to `resolveRelatedModels`.

**Comment by raymondfeng** [comment]
@ritch I renamed the method to be `resolveRelatedModels` to reflect its purpose. We'll revisit it after a new hook is introduced to resolve the model dependencies after all models are defined and registered in the registry.

**Comment by ritch** [comment]
Can you squash and reword the commit so they make sense in the changelog? Also the build is failing.

**Pull request merged**